docs/dataloader.md

```diff
@@ -0,0 +1,5 @@
+# Data Loader
+
+## Background
+
```

Comment:
might be better to break this into paragraphs or separate lines (partly just so easier to comment on details in a PR)

Reply:
we might want to add some additional info on things like that metadata is used to overcome file system limitations, that the dataset loads the data from cos/s3, etc

Reply:
The dataset part was contributed by @daviswer .

@daviswer , can feel free to modify `dataloader.md` in place if you want to add anything here.

Reply:
Didn't realize that writeup was going here! I added pgraph breaks and a pointer to the `dataset_utils.py` file itself for further detail.